lib/matplotlib/path.py

```diff
@@ -280,8 +281,17 @@ def __deepcopy__(self, memo=None):
         Return a deepcopy of the `Path`.  The `Path` will not be
         readonly, even if the source `Path` is.
         """
-        # Deepcopying arrays (vertices, codes) strips the writeable=False flag.
-        p = copy.deepcopy(super(), memo)
+        memo = memo or {}
+        if id(self) in memo:
+            return memo[id(self)]
+
+        cls = type(self)
+        p = cls.__new__(cls)
+        memo[id(self)] = p
```

Comment:
On other thought, updating `memo` may be necessary.

Reply:
Yes, because we recurse into `deepcopy` before returning from this function

Reply:
Thanks for the comments, I rewrote the commit based on them